replica/backup.go

```diff
@@ -204,10 +205,11 @@ func (rb *BackupStatus) CloseSnapshot(snapID, volumeID string) error {
 func (rb *BackupStatus) CompareSnapshot(snapID, compareSnapID, volumeID string) (*backupstore.Mappings, error) {
 	id := GenerateSnapshotDiskName(snapID)
 	compareID := ""
+	rb.lock.Lock()
 	if compareSnapID != "" {
+		rb.IsIncrementalBackup = true
```

Comment:
I thought we can get `IsIncremental` from the backup object. Not sure why we need to set it manually here?

Reply:
There are basically two ways that I know of, for the engine to fetch the `IsIncremental` field and send as a response for backup creation.
1. Create a new API in backupstore library and fetch from the backup object. In this case, I will have to explicitly make two calls - one for getting backupID(CreateDeltaBlockBackup) and another for getting the backup_type(New Api) by using the backupID from the first call.

2. We can learn the backupType as soon as the backupstore invokes `CompareSnapshot` and record it in the `backupStatus` object. 

I chose option 2 as it's simpler and self contained in the replica/backup.go file.

Reply:
We shouldn't need to set `IsIncremental` in two places.

I think it's better for the variable to set during the backup creation process, instead of in the callback, since from the callback we don't know how the caller will deal with the result - for example, some other steps in the later function failed so the caller decides to do a full backup. 

CreateDeltaBlockBackup can return the flag to indicate if it's incremental or not, just like what you did with CreateBackup call.